### PR TITLE
Dr pdc extract study

### PR DESCRIPTION
**Pull request by druth314**

### CODE REVIEW

File: GDC_mapping.yml [hunk 1]
```diff
@@ -19,17 +19,26 @@ ResearchSubject:
 Diagnosis:
     Mapping:
         id: 'cases.diagnoses.diagnosis_id'
+        #identifier:
```
Comment (fkaufman-asym) [inline GDC_mapping.yml:22]: Are we intending to use these in the future?

Comment (druth314) [reply GDC_mapping.yml:22]: I believe we are - I simply want to uncomment and update the hand-written transformation files on another release since it is related to a separate JIRA ticket

File: cdatransform/transform/transform_lib/transform_with_YAML_v1.py [hunk 2]
```diff
@@ -3,102 +3,112 @@
 from cdatransform.transform.validate import LogValidation
```
Comment (fkaufman-asym) [inline cdatransform/transform/transform_lib/transform_with_YAML_v1.py:5]: I'm just gonna ask that maybe when I get back from my vacation that we can do a walkthrough of the transformation code.   I know that we went over this a bit when I first came on the project.  It is a little hard to follow, so I think that doing a screencast of following through this code now that I know the data domain just a bit better.

**Review by fkaufman-asym — COMMENTED** [review]
I'm going to let Dion do an approval on this since his python skill are above mine.  I had some general comments but he and Nemanja might point out a couple of things that I'm not following right away.

File: cdatransform/extract/pdc.py [hunk 3]
```diff
@@ -123,6 +217,26 @@ def get_file_metadata(file_metadata_record) -> dict:
         ]
     }
 
+        
+def agg_cases_info_for_study(study,demo,diag,samp,added_info):
+    out = dict({})
+    out = []
```
[the remark this answers is not in the document]
Comment (druth314) [reply cdatransform/extract/pdc.py:223]: You are right here. I will remove line 222. That was a leftover mistake in the code

File: cdatransform/transform/transform_lib/transform_with_YAML_v1.py (hunk 2 above)
Comment (druth314) [reply cdatransform/transform/transform_lib/transform_with_YAML_v1.py:5]: I agree that it is a bit hard to follow. I added a portion of code that would read from yaml files and some of the repo needs to be cleaned up. There are probably other better ways to organize libraries as well which would make it less confusing. I was going to setup a time with Elijah to go over the transformation code as well so hopefully we can setup a time with all of us

File: PDC_mapping.yml [hunk 4]
```diff
@@ -26,6 +26,9 @@ ResearchSubject:
 Diagnosis:
     Mapping:
         id: 'cases.diagnoses.diagnosis_id'
+        #identifier:
+       #     value: 'cases.diagnoses.diagnosis_id'
+       #     system: 'PDC'
         age_at_diagnosis: 'cases.diagnoses.age_at_diagnosis'
```
[the remark this answers is not in the document]
Comment (druth314) [reply PDC_mapping.yml:32]: Same as above. Going to include fields in a future update

File: cdatransform/transform/transform_lib/transform_with_YAML_v1.py (hunk 2 above)
Comment (fkaufman-asym) [reply cdatransform/transform/transform_lib/transform_with_YAML_v1.py:5]: That sounds great to me.  I added Dion as a reviewer and he's going to go over that right now and will sign off on it for you since I'll be gone next week.

File: cdatransform/transform/transform_lib/transform_with_YAML_v1.py [hunk 5]
```diff
@@ -3,102 +3,112 @@
...
 
 
-def add_Specimen_rec(orig,MandT,DC,**kwargs):
-    cur_path = kwargs.get('cur_path',['cases','samples'])
-    spec_type = kwargs.get('spec_type','samples')
-    rel_path = kwargs.get('rel_path','cases.samples')
+def add_Specimen_rec(orig, MandT, DC, **kwargs):
```
Comment (fkaufman-asym) [inline cdatransform/transform/transform_lib/transform_with_YAML_v1.py:6]: Can we use the variable DataCommons instead of DC across the transform project?  That and ManT just kind of seems not very descriptive to me.

File: cdatransform/extract/pdc.py [hunk 6]
```diff
@@ -123,6 +217,26 @@ def get_file_metadata(file_metadata_record) -> dict:
         ]
     }
 
+        
+def agg_cases_info_for_study(study,demo,diag,samp,added_info):
```
Comment (fkaufman-asym) [inline cdatransform/extract/pdc.py:221]: can we call this sample rather than samp.

File: cdatransform/extract/pdc.py [hunk 7]
```diff
@@ -123,6 +217,26 @@ def get_file_metadata(file_metadata_record) -> dict:
...
+    out = dict({})
+    out = []
+    for demo_case in demo:
+        case_id = demo_case['case_id']
+        demo_case.update(added_info)
+        for diag_ind in range(len(diag)):
+            if diag[diag_ind]['case_id'] == case_id:
+                demo_case['diagnoses'] = diag.pop(diag_ind)['diagnoses']
+                break
+        for samp_ind in range(len(samp)):
```
Comment (fkaufman-asym) [inline cdatransform/extract/pdc.py:231]: I don't think saving a couple fo characters for samp_ind.   I'd rather see sample_idx or  sample_index or even just instance.

File: cdatransform/transform/transform_lib/transform_with_YAML_v1.py (hunk 5 above)
Comment (druth314) [reply cdatransform/transform/transform_lib/transform_with_YAML_v1.py:6]: We can do that. It was just easier to type it all out since they appear very frequently. MandT is short for Mapping and Transform (dictionary)

File: cdatransform/extract/pdc.py (hunk 6 above)
Comment (druth314) [reply cdatransform/extract/pdc.py:221]: Good idea. Will alter and push

File: cdatransform/extract/pdc.py (hunk 7 above)
Comment (druth314) [reply cdatransform/extract/pdc.py:231]: Good idea. Will alter and push

**Review by nemanjab17 — APPROVED** [review]
